flink-runtime/src/main/java/org/apache/flink/runtime/util/bash/BashJavaUtils.java

```diff
@@ -92,9 +102,42 @@ public static void main(String[] args) throws FlinkException {
 		JobManagerProcessSpec jobManagerProcessSpec = JobManagerProcessUtils.processSpecFromConfigWithNewOptionToInterpretLegacyHeap(
 			configuration,
 			JobManagerOptions.JVM_HEAP_MEMORY);
+
+		logMasterConfiguration(jobManagerProcessSpec);
+
 		return Collections.singletonList(ProcessMemoryUtils.generateJvmParametersStr(jobManagerProcessSpec));
 	}
 
+	private static void logMasterConfiguration(JobManagerProcessSpec spec) {
+		JobManagerFlinkMemory flinkMemory = spec.getFlinkMemory();
+		LOG.info("Final Master Memory configuration:");
+		LOG.info("  Total Process Memory: {}", spec.getTotalProcessMemorySize().toHumanReadableString());
+		LOG.info("    Total Flink Memory: {}", flinkMemory.getTotalFlinkMemorySize().toHumanReadableString());
+		LOG.info("      Heap:             {}", flinkMemory.getJvmHeapMemorySize().toHumanReadableString());
+		LOG.info("      Off-heap:         {}", flinkMemory.getJvmDirectMemorySize().toHumanReadableString());
+		LOG.info("    JVM Metaspace:      {}", spec.getJvmMetaspaceSize().toHumanReadableString());
+		LOG.info("    JVM Overhead:       {}", spec.getJvmOverheadSize().toHumanReadableString());
+	}
+
+	private static void logTaskExecutorConfiguration(TaskExecutorProcessSpec spec) {
+		TaskExecutorFlinkMemory flinkMemory = spec.getFlinkMemory();
+		MemorySize totalOffHeapMemory = flinkMemory.getManaged().add(flinkMemory.getJvmDirectMemorySize());
+		LOG.info("Final TaskExecutor Memory configuration:");
```

Comment:
This might be a bit tedious but I think it would be really cool to output this info in the format we use in JavaDocs for `TaskExecutorProcessSpec` if it works. Same for JM.

Reply:
that may be a bit overkill imo; I went more for a concise version.

Reply:
it is the same, just a bit more schematic, ok, maybe it can be even less readable under certain fonts or so.